src/esp/nav/PathFinder.cpp

```diff
@@ -923,16 +923,30 @@ static float frand() {
   return static_cast<float>(rand()) / static_cast<float>(RAND_MAX);
 }
 
-vec3f PathFinder::Impl::getRandomNavigablePoint() {
-  dtPolyRef ref;
-  constexpr float inf = std::numeric_limits<float>::infinity();
-  vec3f pt(inf, inf, inf);
-  dtStatus status =
-      navQuery_->findRandomPoint(filter_.get(), frand, &ref, pt.data());
-  if (!dtStatusSucceed(status)) {
-    LOG(ERROR) << "Failed to getRandomNavigablePoint";
+vec3f PathFinder::Impl::getRandomNavigablePoint(const int maxTries /*= 10*/) {
+  if (getNavigableArea() <= 0.0)
+    throw std::runtime_error(
+        "NavMesh has no navigable area, this indicates an issue with the "
+        "NavMesh");
+
+  vec3f pt;
+
+  int i;
+  for (i = 0; i < maxTries; ++i) {
+    dtPolyRef ref;
+    dtStatus status =
+        navQuery_->findRandomPoint(filter_.get(), frand, &ref, pt.data());
+    if (dtStatusSucceed(status))
+      break;
+  }
+
+  if (i == maxTries) {
+    LOG(ERROR) << "Failed to getRandomNavigablePoint.  Try increasing max "
+                  "tries if the navmesh is fine but just hard to sample from";
+    return {NAN, NAN, NAN};
```

Comment:
FYI there's [Mn::Constants::nan()](https://doc.magnum.graphics/magnum/structMagnum_1_1Math_1_1Constants.html#a0cc96df25e99d8e386551475d6d171fc) in case you wouldn't want to use the macro (mainly suggesting it because it has other common constants as well, all in one place).